drivers/hmis/app/models/hmis/unit.rb

```diff
@@ -162,6 +162,26 @@ def to_pick_list_option
     { code: id, label: display_name }
   end
 
+  # Build an unsaved CE Opportunity for this unit
+  def build_ce_opportunity(rule_resolver: Hmis::Ce::Match::UnitGroupRuleResolver.new)
+    raise 'Unit already has an active opportunity' if latest_opportunity&.active?
+    raise 'Unit must be in a Unit Group to build CE opportunity' unless unit_group
+    raise 'Unit Group has no Workflow Template' unless unit_group.workflow_template || unit_group.direct_referral_workflow_template
+
+    unit_desc = unit_type&.description
+    opportunity_name = "Unit #{id}#{unit_desc ? ' - ' : ''}#{unit_desc}"
+
+    rules = rule_resolver.rules_for_unit_group(unit_group)
+
+    Hmis::Ce::Opportunity.new(
+      unit: self,
+      project: project,
+      name: opportunity_name,
+      candidate_pool_id: unit_group.candidate_pool_id,
```

Comment:
Look at what happens when:
- unit group didn't have a candidate pool when opportunity was minted, and now it does? does the opportunity get marked stale? (i think in practice this happens like ProjectCeConfig updated => candidate pool assigned to unit group => existing opportunity should be stale or add the candidate pool?)